### PR TITLE
Use new let binding for cvc printer

### PR DESCRIPTION
**Pull request by ajreynol**
Also changes names slightly to avoid accidental uses of toStream, which can lead to infinite loops if the wrong version is used.

### CODE REVIEW

File: src/printer/cvc/cvc_printer.cpp [hunk 1]
```diff
@@ -942,17 +924,17 @@ void CvcPrinter::toStream(std::ostream& out,
     // Quantifiers
     case kind::FORALL:
       out << "(FORALL";
-      toStream(out, n[0], depth, false);
+      toStreamNode(out, n[0], depth, true, lbind);
```
Comment (mpreiner) [inline src/printer/cvc/cvc_printer.cpp:927]: Just to make sure that this is not a copy&pase: Should this be really `true`? The original call was `false`.

File: src/printer/cvc/cvc_printer.cpp [hunk 2]
```diff
@@ -942,17 +924,17 @@ void CvcPrinter::toStream(std::ostream& out,
...
       out << " : ";
-      toStream(out, n[1], depth, false);
+      toStreamNodeWithLetify(out, n[1], depth, true, lbind);
```
Comment (mpreiner) [inline src/printer/cvc/cvc_printer.cpp:929]: Ditto.

File: src/printer/cvc/cvc_printer.cpp [hunk 3]
```diff
@@ -942,17 +924,17 @@ void CvcPrinter::toStream(std::ostream& out,
...
       out << ')';
       // TODO: user patterns?
       return;
     case kind::EXISTS:
       out << "(EXISTS";
-      toStream(out, n[0], depth, false);
+      toStreamNode(out, n[0], depth, true, lbind);
```
Comment (mpreiner) [inline src/printer/cvc/cvc_printer.cpp:935]: Ditto.

File: src/printer/cvc/cvc_printer.cpp [hunk 4]
```diff
@@ -942,17 +924,17 @@ void CvcPrinter::toStream(std::ostream& out,
...
       out << " : ";
-      toStream(out, n[1], depth, false);
+      toStreamNodeWithLetify(out, n[1], depth, true, lbind);
```
Comment (mpreiner) [inline src/printer/cvc/cvc_printer.cpp:937]: Ditto.

File: src/printer/cvc/cvc_printer.cpp (hunk 1 above)
Comment (ajreynol) [reply src/printer/cvc/cvc_printer.cpp:927]: Yep, there were inconsistencies which I intentionally fixed here.

**Review by mpreiner — APPROVED** [review]

**Pull request merged**